--- libdesktop-agnostic-0.3.94/libdesktop-agnostic/config-impl-gconf.vala.orig	2024-12-28 18:19:45.717385946 +0100
+++ libdesktop-agnostic-0.3.94/libdesktop-agnostic/config-impl-gconf.vala	2024-12-28 19:11:59.687074441 +0100
@@ -54,7 +54,7 @@ namespace DesktopAgnostic.Config
       Schema schema = this.schema;
 
       this.connection_id = 0;
-      this._notifiers = Datalist<SList<NotifyDelegate>> ();
+      this._notifiers = Datalist<unowned SList<NotifyDelegate>> ();
       base_path = schema.get_metadata_option (opt_prefix +
                                               "base_path").get_string ();
       this.schema_path = "/schemas%s/%s".printf (base_path, schema.app_name);
@@ -324,7 +324,7 @@ namespace DesktopAgnostic.Config
     }
 
     private GLib.ValueArray
-    slist_to_valuearray (SList<unowned GConf.Value> list, Type type) throws Error
+    slist_to_valuearray (SList<GConf.Value> list, Type type) throws Error
     {
       GLib.ValueArray arr = new GLib.ValueArray (list.length ());
       foreach (unowned GConf.Value gc_val in list)
--- libdesktop-agnostic-0.3.92/libdesktop-agnostic/config-notify-delegate.c.orig	2011-03-12 15:14:23.000000000 +0100
+++ libdesktop-agnostic-0.3.92/libdesktop-agnostic/config-notify-delegate.c	2024-12-27 16:43:29.492965924 +0100
@@ -20,7 +20,7 @@
  * Author : Mark Lee <libdesktop-agnostic@lazymalevolence.com>
  */
 
-#include <libdesktop-agnostic/config.h>
+#include <libdesktop-agnostic/desktop-agnostic-cfg.h>
 
 typedef struct _DesktopAgnosticConfigNotifyDelegate
 {
--- libdesktop-agnostic-0.3.94/libdesktop-agnostic/config-schema.vala.orig	2013-11-26 20:04:22.000000000 +0100
+++ libdesktop-agnostic-0.3.94/libdesktop-agnostic/config-schema.vala	2024-12-28 18:19:06.610931137 +0100
@@ -318,7 +318,7 @@ namespace DesktopAgnostic.Config
      * @param group the group name to search for keys associated with it
      * @return a list of zero or more keys
      */
-    public unowned List<unowned string>?
+    public unowned List<string>?
     get_keys (string group)
     {
       return this.keys.lookup (group);
@@ -332,7 +332,7 @@ namespace DesktopAgnostic.Config
     public bool
     exists (string group, string key)
     {
-      unowned List<unowned string> group_keys = this.keys.lookup (group);
+      unowned List<string> group_keys = this.keys.lookup (group);
       return group_keys != null &&
              group_keys.find_custom (key, (CompareFunc)strcmp) != null;
     }
--- libdesktop-agnostic-0.3.92/libdesktop-agnostic/config.vala.orig	2011-03-12 15:14:23.000000000 +0100
+++ libdesktop-agnostic-0.3.92/libdesktop-agnostic/config.vala	2024-12-24 14:01:28.503617417 +0100
@@ -22,7 +22,6 @@
 
 using DesktopAgnostic;
 
-[CCode (cheader_filename = "libdesktop-agnostic/config.h")]
 namespace DesktopAgnostic.Config
 {
   /**
--- libdesktop-agnostic-0.3.92/libdesktop-agnostic/desktop-entry.vala.orig	2011-03-12 15:14:23.000000000 +0100
+++ libdesktop-agnostic-0.3.92/libdesktop-agnostic/desktop-entry.vala	2024-12-24 14:00:48.893832002 +0100
@@ -22,7 +22,6 @@
 
 using DesktopAgnostic;
 
-[CCode (cheader_filename = "libdesktop-agnostic/fdo.h")]
 namespace DesktopAgnostic.FDO
 {
   /**
--- libdesktop-agnostic-0.3.94/libdesktop-agnostic/ui-launcher-editor-dialog.vala.orig	2024-12-28 18:19:45.717385946 +0100
+++ libdesktop-agnostic-0.3.94/libdesktop-agnostic/ui-launcher-editor-dialog.vala	2024-12-28 19:10:20.764277019 +0100
@@ -250,7 +250,7 @@ namespace DesktopAgnostic.UI
       // make sure widgets are properly set up
       on_type_changed (this._type_combo);
 
-      List<unowned Widget> focus_chain_list = new List<unowned Widget> ();
+      List<Widget> focus_chain_list = new List<Widget> ();
       focus_chain_list.append (this._type_combo);
       focus_chain_list.append (this._name);
       focus_chain_list.append (this._desc);
--- libdesktop-agnostic-0.3.94/libdesktop-agnostic/vfs-volume-impl-gio.vala.orig	2013-11-26 20:04:22.000000000 +0100
+++ libdesktop-agnostic-0.3.94/libdesktop-agnostic/vfs-volume-impl-gio.vala	2024-12-28 17:07:32.797526039 +0100
@@ -295,7 +295,7 @@ namespace DesktopAgnostic.VFS
         return (void*)this.monitor;
       }
     }
-    public List<VFS.Volume> volumes
+    public List<weak VFS.Volume> volumes
     {
       owned get
       {
--- libdesktop-agnostic-0.3.92/libdesktop-agnostic/vfs.vala.orig	2011-03-12 15:14:23.000000000 +0100
+++ libdesktop-agnostic-0.3.92/libdesktop-agnostic/vfs.vala	2024-12-24 13:30:57.133538800 +0100
@@ -20,7 +20,6 @@
  * Author : Mark Lee <libdesktop-agnostic@lazymalevolence.com>
  */
 
-[CCode (cheader_filename = "libdesktop-agnostic/vfs.h")]
 namespace DesktopAgnostic.VFS
 {
   public interface Implementation : Object
--- libdesktop-agnostic-0.3.94/libdesktop-agnostic/vfs-volume.vala.orig	2013-11-26 20:04:22.000000000 +0100
+++ libdesktop-agnostic-0.3.94/libdesktop-agnostic/vfs-volume.vala	2024-12-28 17:12:13.346006180 +0100
@@ -61,7 +61,7 @@ namespace DesktopAgnostic.VFS
   public interface VolumeMonitor : Object
   {
     public abstract void* implementation { get; }
-    public abstract List<Volume> volumes { owned get; }
+    public abstract List<weak Volume> volumes { owned get; }
     public abstract signal void volume_mounted (Volume volume);
     public abstract signal void volume_unmounted (Volume volume);
   }
--- libdesktop-agnostic-0.3.92/python/configmodule.c.orig	2011-03-12 15:14:23.000000000 +0100
+++ libdesktop-agnostic-0.3.92/python/configmodule.c	2024-12-27 16:44:05.902356306 +0100
@@ -21,7 +21,7 @@
 #include "build-config.h"
 #endif
 
-#include <libdesktop-agnostic/config.h>
+#include <libdesktop-agnostic/desktop-agnostic-cfg.h>
 
 #include <pygobject.h>
 
--- libdesktop-agnostic-0.3.92/tests/test-vfs-file-monitor.vala.orig	2011-03-12 15:14:23.000000000 +0100
+++ libdesktop-agnostic-0.3.92/tests/test-vfs-file-monitor.vala	2024-12-24 13:13:05.469344503 +0100
@@ -86,7 +86,7 @@ class TestFileMonitor
       unowned string path = args[1];
       file = VFS.file_new_for_path (path);
       monitor = file.monitor ();
-      monitor.changed += on_change;
+      monitor.changed.connect(on_change);
       MainLoop mainloop = new MainLoop (null, false);
       if (args.length == 3 && file.file_type == VFS.FileType.DIRECTORY)
       {
--- libdesktop-agnostic-0.3.92/tests/test-vfs-trash.vala.orig	2011-03-12 15:14:23.000000000 +0100
+++ libdesktop-agnostic-0.3.92/tests/test-vfs-trash.vala	2024-12-24 13:13:28.469219902 +0100
@@ -36,7 +36,7 @@ public class TestTrash
     {
       VFS.init ();
       unowned VFS.Trash t = VFS.trash_get_default ();
-      t.file_count_changed += on_file_count_changed;
+      t.file_count_changed.connect(on_file_count_changed);
       mainloop = new MainLoop (null, true);
       mainloop.run ();
       VFS.shutdown ();
--- libdesktop-agnostic-0.3.94/libdesktop-agnostic/config-impl-keyfile.vala.orig	2024-12-28 22:20:19.022527353 +0100
+++ libdesktop-agnostic-0.3.94/libdesktop-agnostic/config-impl-keyfile.vala	2024-12-29 07:02:30.282791347 +0100
@@ -48,7 +48,7 @@ namespace DesktopAgnostic.Config
       if (this.schema != null)
       {
         this._data = new KeyFile ();
-        this._notifiers = Datalist<SList<NotifyDelegate>> ();
+        this._notifiers = Datalist<unowned SList<NotifyDelegate>> ();
       }
     }
 
--- libdesktop-agnostic-0.3.94/libdesktop-agnostic/desktop-entry-impl-gio.vala.orig	2013-11-26 20:04:22.000000000 +0100
+++ libdesktop-agnostic-0.3.94/libdesktop-agnostic/desktop-entry-impl-gio.vala	2024-12-29 07:04:19.025535570 +0100
@@ -322,8 +322,8 @@ namespace DesktopAgnostic.FDO
     launch (DesktopEntryLaunchFlags flags,
             SList<string>? documents) throws GLib.Error
     {
-      List<unowned string> uris = new List<unowned string> ();
-      foreach (unowned string s in documents)
+      List<string> uris = new List<string> ();
+      foreach (string s in documents)
       {
         uris.append (s);
       }
